cunumeric/module.py

```diff
@@ -523,6 +523,105 @@ def tile(a, reps):
     return result
 
 
+# Spliting arrays
+@copy_docstring(np.vsplit)
+def vsplit(a, indices):
+    return split(a, indices, axis=0)
+
+
+@copy_docstring(np.hsplit)
+def hsplit(a, indices):
+    return split(a, indices, axis=1)
+
+
+@copy_docstring(np.dsplit)
+def dsplit(a, indices):
+    return split(a, indices, axis=2)
+
+
+@copy_docstring(np.split)
+def split(a, indices, axis=0):
+    return array_split(a, indices, axis, equal=True)
+
+
+@copy_docstring(np.array_split)
+def array_split(a, indices, axis=0, equal=False):
+    array = ndarray.convert_to_cunumeric_ndarray(a)
+    dtype = type(indices)
+    split_pts = []
+    if axis >= array.ndim:
+        raise ValueError(
+            f"array({array.shape}) has less dimensions than axis({axis})"
+        )
+
+    if dtype == int:
+        res = array.shape[axis] % indices
+        if equal and res != 0:
+            raise ValueError("array split does not result in an equal divison")
+
+        len_subarr = array.shape[axis] // indices
+        end_idx = array.shape[axis]
+        first_idx = len_subarr
+
+        # the requested # of subarray is larger than the size of array
+        # -> size of 1 subarrays + empty subarrays
+        if len_subarr == 0:
+            len_subarr = 1
+            first_idx = len_subarr
+            end_idx = indices
+        else:
+            if res != 0:
+                # The first 'res' groups have len_subarr+1 elements
+                split_pts = list(
+                    range(
+                        len_subarr + 1, (len_subarr + 1) * res, len_subarr + 1
+                    )
+                )
+                first_idx = (len_subarr + 1) * res
+        split_pts.extend(range(first_idx, end_idx + 1, len_subarr))
+    elif dtype == np.array or dtype == list or dtype == tuple:
```

Comment:
In the case where `indices` is a NumPy array also check that it's an integer array (i.e. `indices.dtype == int`)